### PR TITLE
Slew correction

### PR DESCRIPTION
**Pull request by AllaMaevskaya**

### CODE REVIEW

File: Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CalibrationInfoObject.h [hunk 1]
```diff
@@ -13,12 +13,14 @@
 
 #include "Rtypes.h"
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CalibrationInfoObject.h:14]: Can this go to DataFormats?

File: Detectors/FIT/FT0/calibration/testWorkflow/slew_upload.cxx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  swTot.Start();
+
+  o2::ft0::FT0CalibTimeSlewing sl;
+  sl.setSingleFileName(inFileName);
+  sl.setMergedFileName(mergedFileName);
+  sl.setNfiles(nFiles);
+  sl.mergeFilesWithTree();
+  for (int iCh = 0; iCh < o2::ft0 ::Geometry::Nchannels; ++iCh) {
+    TH2F* hist = sl.getTimeAmpHist(iCh);
+    if (hist->GetEntries() < 100)
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/testWorkflow/slew_upload.cxx]: {} missing

File: Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h [hunk 3]
```diff
@@ -0,0 +1,117 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h:1]: I don't see corresponding cxx file and it is not in the CMakeFiles. Looks like a stray file.

File: Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  if (mTFsendingPolicy || c->getCollectedCalibInfoSlot().size() > maxNumberOfHits) {
+    return true;
+  }
+  return false;
+}
+
+//_____________________________________________
+void FT0CalibCollector::finalizeSlot(Slot& slot)
+{
+  // here we fill the tree with the remaining stuff that was not filled before
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx]: I don't see any tree filled here. As far as I understand, the mFT0CollectedCalibInfo is sent to the writer, which writes the tree?
When this tree should be processed (by slew_upload.cxx  ?) ?
For how long are you going to accumulate these tree(s)?

File: Detectors/FIT/FT0/calibration/src/FT0CalibTimeSlewing.cxx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+//______________________________________________
+void FT0CalibTimeSlewing::mergeFilesWithTree()
+{
+  TFileMerger merger;
+  merger.OutputFile(mMergedFileName.c_str());
+  for (Int_t i = 0; i < mNfiles; i++) {
+    TFile* file =
+      TFile::Open(Form("%s_%d.root", mSingleFileName.c_str(), i));
+    if (file)
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/src/FT0CalibTimeSlewing.cxx]: {} missing, also on the next line

File: Detectors/FIT/FT0/calibration/testWorkflow/slew_upload.cxx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file digi2raw.cxx
+/// \author ruben.shahoyan@cern.ch
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/testWorkflow/slew_upload.cxx]: fix the filename and the author, please

File: Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+  o2::ft0::FT0CalibInfoSlot* c = slot.getContainer();
+  mFT0CollectedCalibInfo = c->getCollectedCalibInfoSlot();
```
Comment (shahor02) [inline Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx:146]: How your slot is defined? Do you have many or just 1? Note that in principle, with multiple slots the TimeSlotCalibration may finalize multiple slots while processing a single TF (e.g. if there was pause, the newly arrived TF may trigger finalization of multiple slots). In this case you will store to the writer only the last slot.

File: Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CalibrationInfoObject.h (hunk 1 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CalibrationInfoObject.h:14]: Yes, sure, but with next improvement, OK?

File: Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx (hunk 4 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx]: these threes could be accumulate ~ 1 day of Pb-Pb data taking and then processed by slew_upload.cxx.

File: Detectors/FIT/FT0/calibration/testWorkflow/slew_upload.cxx (hunk 6 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/testWorkflow/slew_upload.cxx]: Oooo, sorry

File: Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CalibrationInfoObject.h (hunk 1 above)
Comment (shahor02) [reply Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CalibrationInfoObject.h:14]: ok, but please don't forget, all data types should eventually move to DataFormats

File: Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx (hunk 4 above)
Comment (shahor02) [reply Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx]: Is this a regular procedure or you do it only once? Do you have an estimate of the real beam time needed for this calibration?
1 day of pbpb is a hell of data, if this is not a 1 time calibration and single run data will be sufficient, perhaps the processing can be done directly from the Collector device?

Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx]: We will do it once. And we need at least 500 entries in all amplitude bins, 1 run is not enough. During Run2 we merged runs for 2-3 days

Comment (shahor02) [reply Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx]: ok, thanks! For single upload the manual operation is ok, but just for the record: Statistic of 2-3 days in Run2 corresponds to <1h in Run3.

File: Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h (hunk 3 above)
Comment (shahor02) [reply Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h:1]: Hi @AllaMaevskaya 
Did you miss this comment?

File: Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx (hunk 4 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx]: We can stop and check after 1 hour data collection - do we have enough entries at high amplitudes.

Comment (shahor02) [reply Detectors/FIT/FT0/calibration/src/FT0CalibCollector.cxx]: @AllaMaevskaya in principle, you should be able to estimate this from the MC. 
But I was asking about https://github.com/AliceO2Group/AliceO2/pull/6403#discussion_r650414955 comment: looks like  Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h  is a stray file

File: Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h (hunk 3 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h:1]: This file should be removed.

Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/calibration/include/FT0Calibration/FT0CollectCalibInfo.h:1]: this file was removed from git

**Review by shahor02 — APPROVED** [review]

**Pull request merged**